smoothscroll.js

```diff
@@ -1,156 +1,237 @@
-(function () {
+(function (w, doc, undefined) {
   'use strict';
 
-  if ('scrollBehavior' in document.documentElement.style) return;
-
-  // TODO: make this intelligent according to distance.
-  var SCROLL_TIME = 300;
-
-  var originalScrollTo = window.scrollTo;
-  var originalScrollBy = window.scrollBy;
-  var originalScrollIntoView = Element.prototype.scrollIntoView;
-
-  // store generally accessible frame id in case a new scroll animation is triggered before the previous
-  // completes, we can cancel the previous scroll.
-  var frame;
-
-  var startY, startX, endX, endY;
-
+  /*
+   * alias
+   * w: window global object
+   * doc: document
+   * undefined: undefined
+   */
+
+  // return if scrollBehavior is supported
+  if ('scrollBehavior' in doc.documentElement.style) return;
+
+  var SCROLL_TIME = 400,
+      // legacy scrolling methods
+      originalScrollTo = w.scrollTo,
+      originalScrollBy = w.scrollBy,
+      originalScrollIntoView = w.Element.prototype.scrollIntoView,
+      // global frame variable to avoid collision
+      frame,
+      // global metric variables
+      startY, startX, endX, endY;
```

Comment:
Not on the ```window``` object, but it could be possible that browsers allow scrolling in the window and an element, but I would have to test. Either way, functionality was not extended or modified.